src/main.rs

```diff
@@ -5,27 +5,45 @@ mod plot;
 mod popup;
 mod table;
 use gui::NWBView;
+use image::GenericImageView;
 
 fn main() {
-    let icon = image::open("static/icon.png")
-        .expect("Failed to open icon path")
-        .to_rgba8();
-    let (icon_width, icon_height) = icon.dimensions();
+    const ICON: &[u8] = include_bytes!("../static/icon.png");
 
-    let options = eframe::NativeOptions {
-        icon_data: Some(eframe::IconData {
-            rgba: icon.into_raw(),
-            width: icon_width,
-            height: icon_height,
-        }),
-        drag_and_drop_support: true,
-        ..Default::default()
-    };
+    match image::load_from_memory(ICON) {
+        Ok(icon) => {
+            let (icon_width, icon_height) = icon.dimensions();
 
-    eframe::run_native(
-        "NWB View",
-        options,
-        Box::new(|_cc| Box::<NWBView>::default()),
-    )
-    .ok();
+            let options = eframe::NativeOptions {
+                icon_data: Some(eframe::IconData {
+                    rgba: icon.into_rgba8().to_vec(),
+                    width: icon_width,
+                    height: icon_height,
+                }),
+                drag_and_drop_support: true,
+                ..Default::default()
+            };
+
+            eframe::run_native(
+                "NWB View",
+                options,
+                Box::new(|_cc| Box::<NWBView>::default()),
+            )
+            .ok();
+        }
+        Err(error) => {
+            println!("Error raised while loading icon: {}", error);
+            println!("Launching without icon...");
+            let options = eframe::NativeOptions {
+                drag_and_drop_support: true,
+                ..Default::default()
+            };
+            eframe::run_native(
+                "NWB View",
+                options,
+                Box::new(|_cc| Box::<NWBView>::default()),
+            )
+            .ok();
+        }
+    }
```

Comment:
Just wondering, isn't it possible to write something like:
```rust
let options = match image::load_from_memory(ICON) {
    Ok(icon) => eframe::NativeOptions{
        // stuff with the icon
    },
    Err(error) => {
        // print the error
        eframe::NativeOptions{
        // stuff without the icon
    }
}
eframe::run_native(
// ....
)
```

Reply:
Yes definitely!